woocommerce/woocommerce-ios
# Domain settings: feature flag, entry point from settings, barebone UI

Part of #8558 


## Description


### Why

After the user creates a store, they might want to choose a proper domain for their new store. This is part of the onboarding plan for the future, until then we're showing the entry point in Menu tab > Settings > Store settings.

### Implementation

This is the first PR of the domain settings, with a feature flag, an entry point in the settings, and a basic UI with mock data. The entry point is only shown when the feature is enabled and the site is a WPCOM site. Some barebone data models in the Networking layer were created to streamline the UI integration, and the Yosemite actions just return mock data for now.

#### Data layer

Two endpoints are required to show the UI on this screen:
- `DomainAction.loadDomains`: this loads the site's domains, including the free staging domain
- `PaymentAction.loadSiteCurrentPlan`: this loads the site's WPCOM plans, and returns the current plan to determine if the site has domain credit so that the user can pick a domain for free (available from certain WPCOM plans)

#### UI

The main SwiftUI view is `DomainSettingsView`, which shows the free `*.wpcomstaging.com` domain at the top with `FreeStagingDomainView`. Below the free domain, different content can be shown with the following logic:
- If the site has a domain credit, a domain credit section is shown for the user to pick a domain for free using their domain credit (not implemented in this PR)
- If the site has existing domains other than the free staging domain, a list of domains is shown. Otherwise, a CTA `Search for a domain` is shown (the CTA is implemented in this PR without the details text below)

## Testing instructions


Prerequisite: a WC store with a WPCOM plan is required

- Log in if needed, and continue with a WPCOM store
- Go to the Menu tab
- Tap on the settings CTA --> there should be a `Domain` row under the store settings section
- Tap the `Domain` row --> a modal should be shown with a free staging domain at the top (mock data) and a CTA at the bottom

## Screenshots


light | dark
-- | --
![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-09 at 10 54 59](https://user-images.githubusercontent.com/1945542/211234228-a8674699-3268-4325-9875-83514aad122a.png) | ![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-09 at 10 55 06](https://user-images.githubusercontent.com/1945542/211234232-5014fceb-6bce-4d87-aefa-c6cf75d3dc9c.png)


---
